### PR TITLE
Adapt settings bundle to new data model

### PR DESCRIPTION
**Pull request by kulmann**
DO NOT MERGE. This is work in progress.

Data model for settings and settings bundles has changed. This PR adapts settings bundles in ocis-accounts to the most recent version. Until ocis-settings is released, you have to checkout https://github.com/owncloud/ocis-settings/pull/35 locally and make a local replace in `go.mod`.

### CODE REVIEW

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/v0/timezones.go  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5833923&bid=19101733)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/v0/settings.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5833923&bid=19101733)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/v0/settings.go  2
         

Clones added
============
- pkg/service/v0/settings.go  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5833923&bid=19101733)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/v0/settings.go  2
         

Clones added
============
- pkg/service/v0/settings.go  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5833923&bid=19101733)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/v0/settings.go  2
         

Clones added
============
- pkg/service/v0/settings.go  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5833923&bid=19101733)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/v0/settings.go  2
         

Clones added
============
- pkg/service/v0/settings.go  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5833923&bid=19101733)

**Comment by refs** [comment]
bug: with this changes, starting the ocis-accounts service without the ocis-settings blows up:

```
go run cmd/ocis-accounts/main.go server
2020-07-20T16:52:36+02:00 INF Starting server addr=0.0.0.0:9181 service=accounts transport=http
2020-07-20T16:52:36+02:00 INF Starting server addr=0.0.0.0:9180 service=accounts transport=grpc
2020-07-20T16:52:37+02:00 INF Reporting settings bundles to settings service service=com.owncloud.api.accounts
2020-07-20 16:52:37  file=service.go:206 level=info Starting [service] com.owncloud.api.accounts
2020-07-20 16:52:37  file=grpc.go:845 level=info Server [grpc] Listening on [::]:9180
2020-07-20 16:52:37  file=service.go:239 level=info Listening on [::]:9181
2020-07-20 16:52:37  file=grpc.go:676 level=info Registry [mdns] Registering node: com.owncloud.api.accounts-4e32790c-0b1e-4fbd-8b17-2297b3f61aa1
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x49f1b8c]

goroutine 56 [running]:
github.com/owncloud/ocis-accounts/pkg/service/v0.RegisterSettingsBundles(0xc000402a80)
        /Users/aunger/code/owncloud/ocis-accounts/pkg/service/v0/settings.go:29 +0x3dc
created by github.com/owncloud/ocis-accounts/pkg/command.Server.func2.2
        /Users/aunger/code/owncloud/ocis-accounts/pkg/command/server.go:77 +0xd4
exit status 2
```

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/v0/settings.go  2
         

Clones added
============
- pkg/service/v0/settings.go  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5833923&bid=19101733)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/v0/settings.go  2
         

Clones added
============
- pkg/service/v0/settings.go  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5833923&bid=19101733)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/v0/settings.go  2
         

Clones added
============
- pkg/service/v0/settings.go  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5833923&bid=19101733)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- ui/helpers/utils.js  2
- pkg/service/v0/settings.go  2
         

Clones added
============
- pkg/service/v0/settings.go  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5833923&bid=19101733)

**Comment by LukasHirt** [comment]
Basic UI done. Just need to find a nice way to use the swagger generated APIs from settings instead of `fetch`.

image
image

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/v0/settings.go  2
- ui/helpers/utils.js  2
- ui/helpers/auth.js  3
         

Complexity decreasing per file
==============================
+ ui/store/index.js  -1
         

Clones added
============
- pkg/service/v0/settings.go  5
- ui/client/settings/index.js  84
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5833923&bid=19101733)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/v0/settings.go  2
- ui/helpers/utils.js  2
- ui/helpers/auth.js  3
         

Complexity decreasing per file
==============================
+ ui/store/index.js  -1
         

Clones added
============
- ui/client/settings/index.js  84
- pkg/service/v0/settings.go  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5833923&bid=19101733)

File: ui/components/accounts/AccountsListRow.vue [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        :drop-id="`accounts-roles-select-dropdown-${account.id}`"
+        :toggle="`.accounts-roles-select-trigger-${account.id}`"
+        mode="click"
+        close-on-click
+        :options="{ delayHide: 0 }"
+      >
+        <ul class="uk-nav">
+          <li v-for="role in roles" :key="role.id" class="uk-margin-small">
+            <oc-button variation="raw" @click="changeRole(role.id)">
+              <span v-text="role.displayName" :class="{ 'uk-text-bold': role === currentRole }" />
```
Comment (kulmann) [inline ui/components/accounts/AccountsListRow.vue:26]: the only thing I can find for improvement is, that the drop could use a visual indicator for the already selected role - not just the bold text, but e.g. a clicked radiobutton. My original idea a while ago was, to bring all the form elements from ocis-settings into the ODS and reuse them all over the place. In ocis-settings there is a single choice dropdown with radiobuttons and a multi choice dropdown with checkboxes. Maybe it's possible to reuse the single choice dropdown with radiobuttons here, and then later on bring a more generic implementation into ODS and use it whereever possible.
See [ocis-settings single-choice](https://github.com/owncloud/ocis-settings/blob/master/ui/components/settings/SettingSingleChoice.vue) for reference.

Comment (kulmann) [reply ui/components/accounts/AccountsListRow.vue:26]: @LukasHirt ^

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- ui/helpers/utils.js  2
- pkg/service/v0/settings.go  2
- ui/helpers/auth.js  3
         

Complexity decreasing per file
==============================
+ ui/store/index.js  -1
         

Clones added
============
- ui/client/settings/index.js  84
- pkg/service/v0/settings.go  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5833923&bid=19101733)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- ui/helpers/utils.js  2
- pkg/service/v0/settings.go  2
- ui/helpers/auth.js  3
         

Complexity decreasing per file
==============================
+ ui/store/index.js  -1
         

Clones added
============
- ui/client/settings/index.js  84
- pkg/service/v0/settings.go  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5833923&bid=19101733)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- ui/helpers/auth.js  3
- pkg/service/v0/settings.go  2
- ui/helpers/utils.js  2
         

Complexity decreasing per file
==============================
+ ui/store/index.js  -1
         

Clones added
============
- ui/client/settings/index.js  84
- pkg/service/v0/settings.go  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5833923&bid=19101733)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/v0/settings.go  2
- ui/helpers/utils.js  2
- ui/helpers/auth.js  3
         

Complexity decreasing per file
==============================
+ ui/store/index.js  -1
         

Clones added
============
- pkg/service/v0/settings.go  5
- ui/client/settings/index.js  84
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5833923&bid=19101733)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- ui/helpers/auth.js  3
- ui/helpers/utils.js  2
- pkg/service/v0/settings.go  2
         

Complexity decreasing per file
==============================
+ ui/store/index.js  -1
         

Clones added
============
- ui/client/settings/index.js  84
- pkg/service/v0/settings.go  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5833923&bid=19101733)

**Comment by kulmann** [comment]
Superseded by https://github.com/owncloud/ocis-accounts/pull/87, https://github.com/owncloud/ocis-accounts/pull/88 and https://github.com/owncloud/ocis-accounts/pull/89

**Pull request closed without merging**